src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,322 +76,548 @@ private static VersionInfo GetVersionInfo()
+            };
+            return cols;
         }
 
-        public static NAIndicatorTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        // Factory method for SignatureLoadModel
+        internal static NAIndicatorTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
             ctx.CheckAtModel(GetVersionInfo());
-            return h.Apply("Loading Model", ch => new NAIndicatorTransform(h, ctx, input));
+
+            return new NAIndicatorTransform(env, ctx);
         }
 
+        // Factory method for SignatureDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+            => new NAIndicatorTransform(env, args).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        internal static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        /// <summary>
+        /// Saves the transform.
+        /// </summary>
         public override void Save(ModelSaveContext ctx)
         {
             Host.CheckValue(ctx, nameof(ctx));
+
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
 
-            // *** Binary format ***
-            // <base>
-            SaveBase(ctx);
+            SaveColumns(ctx);
         }
 
-        private ColumnType[] GetTypesAndMetadata()
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+            => new Mapper(this, schema);
+
+        private sealed class Mapper : MapperBase
         {
-            var md = Metadata;
-            var types = new ColumnType[Infos.Length];
-            for (int iinfo = 0; iinfo < Infos.Length; iinfo++)
+            private sealed class ColInfo
+            {
+                public readonly string Output;
+                public readonly string Input;
+                public readonly ColumnType OutputType;
+                public readonly ColumnType InputType;
+                public readonly Delegate InputIsNA;
+
+                public ColInfo(string input, string output, ColumnType inType, ColumnType outType)
+                {
+                    Input = input;
+                    Output = output;
+                    InputType = inType;
+                    OutputType = outType;
+                    InputIsNA = GetIsNADelegate(InputType); ;
+                }
+            }
+
+            // are we sure we need this? maybe not?
```

Comment:
Cant this comment be removed? 
Also can the members be grouped above the constructor with the other fields?